Update editable dropdown tooltip text with new selected values

Charles-Gagnon opened the pull request:
Fix our custom editable dropdown control to update the tooltip text correctly when a new value is selected. Previous behavior was to always keep the initial text (<Default> for example). Now we'll update it as appropriate (and default back to Placeholder text when we're clearing the value)

Charles-Gagnon commented:
Hm, that's a fair point. But what about really long text? It's nice to have the tooltip display with the full name in that case so it isn't cut off :

![image](https://user-images.githubusercontent.com/28519865/54455536-4d691e80-4719-11e9-8d31-a76e76a3a378.png)

MattIrv commented:
Looking around online a bit it seems like most dropdowns don't have title text at all. I do like the idea of keeping it the selected value in case that value is too long to show. Is it also possible for values to be too long to show when selecting them from the drop down? If so I'd recomment adding title text for those as well.

Are there other inputs where we show a description on hover?

Charles-Gagnon commented:
Actually it looks like we already have that 

![image](https://user-images.githubusercontent.com/28519865/54465242-6d0f3f80-4737-11e9-9da2-4e1d3c35adc0.png)

With that I'm thinking we just don't show a tooltip for the component itself at all. None of the other dropdowns do and it doesn't seem super useful since the label is right there.

Charles-Gagnon commented:
Ok, changed this now so we don't show the tooltips for the actual text box part. We'll still show the tooltips for the actual entries in the list though 

![image](https://user-images.githubusercontent.com/28519865/54631639-a8608580-4a39-11e9-949f-3291271a6722.png)

I also removed the tooltips from the arrows - they didn't make any sense being there and generally didn't seem super useful. I opted to keep the value in the IDropdownOptions though in case we want to actually use it in the future (although I'm not opposed to just removing it completely since I kind of doubt this would ever be a valid use case)

MattIrv commented on src/sql/base/browser/ui/editableDropdown/dropdown.ts:
```diff
@@ -70,7 +70,7 @@ const defaults: IDropdownOptions = {
 	maxHeight: 300,
 	errorMessage: errorMessage,
 	contextBorder: Color.fromHex('#696969'),
-	actionLabel: nls.localize('dropdownAction.toggle', "Toggle dropdown")
```
What are these `actionLabel`s for? Were they added for accessibility reasons?

MattIrv replied:
Looking at the git blame for these they were definitely added in an accessibility commit by @anthonydresser. Anthony do you remember if we need these labels for accessibility? I'd recommend keeping them so we don't accidentally reintroduce accessibility bugs.

Charles-Gagnon replied:
We should probably be using the aria-label for that instead then - the tooltip showing up is weird and not normal for dropdowns.

anthonydresser replied:
Yeah you shouldn't remove this; this is an accessibility requirement for screen readers that the down arrow states its action.

Charles-Gagnon replied:
But there shouldn't be any issue with using aria-label right? That's what we use in most of the rest of the product for accessibility so using the tooltip here seems odd.

Charles-Gagnon replied:
Alright, so testing this out the ideal solution is to use aria-label. Unfortunately the actions are not hooked up in a way that we can directly add that attribute - it's a separate VS class that creates the actual elements and it doesn't support aria-label. I could do some DOM manipulation to set it but I'd avoid doing hacky stuff like that - especially since it works fine right now (it's just a little odd to have the tooltip there).

anthonydresser commented on src/sql/base/browser/ui/editableDropdown/dropdown.ts:
```diff
@@ -129,6 +129,10 @@ export class Dropdown extends Disposable {
+		// Clear title from input box element (defaults to placeholder value) since we don't want a tooltip for the selected value
+		// in the text box - we already have tooltips for each item in the dropdown itself.
+		this._input.inputElement.title = '';
+
```
Whats the reason for not wanting the title to be the placeholder? thats standard across vscode since the placeholder should be a description of the input.

Charles-Gagnon replied:
The issue is how we're using it for database dropdowns. The placeholder value in that case is <default> - which doesn't make sense as a tooltip (especially after they've chosen an actual database)

We could have the title be set to some different value for that case though for accessibility - "Select a database" or something.

Charles-Gagnon replied:
We correctly set the aria-label to "Database" so this still works correctly with the screen reader. We just no longer have the tooltip with "<default>" which is what I was originally going for.

MattIrv replied:
It makes sense if the placeholder is actually an unselectable placeholder that describes the drop down but doesn't work for drop downs where there's a default selection and not a placeholder. Maybe there's some way to differentiate them? But unless this is breaking any existing dropdowns where we rely on that functionality the change seems good to me
